File: shop-heroes-v2/src/app/objects/recipe.ts
import { Material } from "./material";
import { Category } from "./category";

export class Recipe {

    constructor(name: string, golds: number, xp: number, minLevel: number, materials: Material[], category: Category){
        this.name = name;
        this.golds = golds;
        this.xp = xp;
        this.minLevel = minLevel;
        this.materials = materials;
        this.category = category;
    }

    id: number;
    name: string;
    golds: number;
    xp: number;
    minLevel: number;
    materials: Material[];
    category: Category;

}

File: ShopHeroesV2/src/main/java/apiRest/database/classe/Recipe.java
package apiRest.database.classe;

import java.util.List;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class Recipe {
	
	private long id;
	private String name;
	private Category category;
	private int minLevel;
	private int golds;
	private int xp;
	private Job job;
	private List<Material> materials;

}
